azure-ipam/ipam.go

```diff
@@ -85,17 +106,25 @@ func (p *IPAMPlugin) CmdAdd(args *cniSkel.CmdArgs) error {
 		p.logger.Error("Failed to interpret CNS IPConfigResponse", zap.Error(err), zap.Any("response", resp))
 		return cniTypes.NewError(ErrProcessIPConfigResponse, err.Error(), "failed to interpret CNS IPConfigResponse")
 	}
-	p.logger.Debug("Parsed pod IP", zap.String("podIPNet", podIPNet.String()))
-
-	cniResult := &types100.Result{
-		IPs: []*types100.IPConfig{
-			{
-				Address: net.IPNet{
-					IP:   net.ParseIP(podIPNet.Addr().String()),
-					Mask: net.CIDRMask(podIPNet.Bits(), 32), // nolint
-				},
-			},
-		},
+	for _, IPNet := range *podIPNet {
+		p.logger.Debug("Parsed pod IP", zap.String("podIPNet", IPNet.String()))
+	}
+
+	cniResult := &types100.Result{}
+	for _, IPNet := range *podIPNet {
+		ipConfig := &types100.IPConfig{}
+		if net.ParseIP(IPNet.Addr().String()).To4() != nil {
+			ipConfig.Address = net.IPNet{
+				IP:   net.ParseIP(IPNet.Addr().String()),
+				Mask: net.CIDRMask(IPNet.Bits(), 32), // nolint
+			}
+		} else {
+			ipConfig.Address = net.IPNet{
+				IP:   net.ParseIP(IPNet.Addr().String()),
+				Mask: net.CIDRMask(IPNet.Bits(), 128), // nolint
+			}
+		}
+		cniResult.IPs = append(cniResult.IPs, ipConfig)
```

Comment:
instead of appending, if you know the length of your slice ahead of time (which I think you do in this case), you can just `make` the slice with the correct length first and insert directly into the slice via indexing. Since this slice will only ever be lengths 0, 1, or 2, it's not a big deal, but something to get into the habit of.

Reply:
Added a `make(...)` for `cniResult.IPs`. I'm guessing the purpose of this is to strictly claim the exact amount of memory you need instead of having a slice with an undefined length.